v2/apierror/apierror.go

```diff
@@ -60,6 +61,23 @@ type ErrDetails struct {
 	Unknown []interface{}
 }
 
+// ExtractMessage provides a mechanism for extracting messages from the Unknown
+// error details.  An entry matching the type of exampleMesg is returned
+// if present.
+func (e ErrDetails) ExtractMessage(exampleMesg interface{}) interface{} {
+	if exampleMesg == nil {
+		return nil
+	}
+	typ := reflect.TypeOf(exampleMesg)
+	for _, elem := range e.Unknown {
+		elemType := reflect.TypeOf(elem)
+		if typ == elemType {
+			return elem
```

Comment:
I don't know why `Unknown` is a slice, but is there a possibility of more than one match here? Should this method collect all matches and return instead `[]interface{}`?

Reply:
In practice, I haven't encountered an example of an API returning multiple details of the same type, nor have I really seen examples of multiple details in general.  I think it was largely built that way for future proofing.  One way to handle this might be to leave this method for the single case, and possibly add an ExtractAll or similar that can return multiple entries.  I'd love to understand if this is common.

Reply:
sgtm